docs/language_guide.md

```diff
@@ -0,0 +1,619 @@
+# Language Guide
+
+Sysl is a system specification language (and toolkit). You describe applications and data models that make up your system and use `sysl` commands like `sd`, `ints` and `data` to render sequence, integration and data diagrams. Sysl can be extended using plugins. `reljam` is a plugin that reads intermediate representation generated by `sysl` to generate Spring based Java RestController for your API defined in sysl. `gosysl` is another plugin that generates `go` stubs for the same api definition.
+
+Your complete system can be described as forest of trees, where one tree represents one application or data model.
+
+Sysl uses indentation to represent parent child or `has` relationships. E.g. an  `application` has `endpoints` or a `table` has `columns`.
+
+## Concepts
+Sysl allows you to specify [Application](#application) behaviour and [Data Model](#data) that is shared between your applications. Another related concept is of software [Projects](#projects) where you can document what changes happened in each project or a release.
+
+More on this below.
+
+### Application
+An application is an independent entity that provides services via its various `endpoints`. E.g. a `MobileApp` allows users to `Login`, `Search` or `Order` items.
+
+Let's start to define a mobile app with these high level endpoints (or flows):
+```
+MobileApp:
```

Comment:
if `MobileApp` happens to be an application, could we make it clearer that this top-level thing is an application?